mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -927,6 +927,7 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
     require(intermediateRDDStorageLevel != StorageLevel.NONE,
       "ALS is not designed to run without persisting intermediate RDDs.")
 
+    ratings.persist()
```

Comment:
Here also persist is not required. And `blockRatings` which is a child RDD is already persisted.

Reply:
You might ignore this: at line 926, ratings.isEmpty() applies an action take() on ratings.